cmd/agent/daemon/state/prometheus_netflow_exporter.go

```diff
@@ -0,0 +1,194 @@
+package state
+
+import (
+	"context"
+	"net/netip"
+
+	castaipb "github.com/castai/kvisor/api/v1/runtime"
+	"github.com/castai/kvisor/cmd/agent/daemon/metrics"
+	"github.com/castai/kvisor/pkg/logging"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	// egressd-compatible metrics
+	egressdTransmitBytesTotal = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "egressd_transmit_bytes_total",
```

Comment:
Sadly this approach will not work efficiently. In egressd we have explicit metrics push via remote write api.
https://github.com/castai/egressd/blob/main/exporter/sinks/prom.go#L116

By using remote write api it will support not only original prometheus but mimir/cortex too. Also it allows to have different prometheus instance just for netflows metrics.

Reply:
On egressd was the controller gathering metrics from the agent and then pushing remotely.

Probably kvisor has to implement the same approach... Do you have it on the roadmap?

Reply:
Yes, kvisor agent could push metrics directly from each agent to prometheus similar to clickhouse exporter.
For now we are no plans for it, but if you could update current prometheus_netflow_exporter that would great :)